### PR TITLE
Shopify i18n

### LINKED ISSUE #314: Shopify translation not working
In order to get the Shopify product translations, we have to add the Accept-Language Header to the API request.
I did this in the fetch-graphql-api.ts file.

`const res = await fetch(API_URL, {
    ...fetchOptions,
    method: 'POST',
    headers: {
      'Accept-Language': 'en-US',
      'X-Shopify-Storefront-Access-Token': API_TOKEN!,
      ...fetchOptions?.headers,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query,
      variables,
    }),
  })`

I can hardcode the language (en-US or es) and it works well. I am getting all my translations.
But how can I get this dynamically connected with the I18nWidget?
Out of the box, the Accept-Language Header is not set and the I18nWidget does not work.

I tried to get the current local from the useRouter hook but I am not able to use the hook in the fetch-graphql-api.ts file

`
const fetchGraphqlApi: GraphQLFetcher = async (
  query: string,
  { variables } = {},
  fetchOptions
) => {
  
  const { locale } = useRouter()
  console.log(locale)

  const res = await fetch(API_URL, {
    ...fetchOptions,
    method: 'POST',
    headers: {
      'Accept-Language': 'en-US',
      'X-Shopify-Storefront-Access-Token': API_TOKEN!,
      ...fetchOptions?.headers,
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      query,
      variables,
    }),
  })
`

But apparently, hooks cannot be used in that file.
Error: Invalid hook call. Hooks can only be called inside of the body of a function component

What is the best way to get Shopify translations working?

Thanks for your help and tips in advance.

### PR DESCRIPTION
**Pull request by cond0r**
Fixes #314 #315

In order to have translated content on Shopify, an app like https://apps.shopify.com/content-translation that automatically / manually translates the content, is required .

https://commerce-gfiys1cog-cond0r.vercel.app/es/product/vans-authentic-lo-pro-burgandy-white

### CODE REVIEW

**Comment by okbel** [comment]
Tiny merge conflict @cond0r

**Comment by cond0r** [comment]
@okbel resolved :)

File: lib/colors.ts [hunk 1]
```diff
@@ -56,6 +56,8 @@ const colorMap: Record<string, string> = {
   blueviolet: '#8A2BE2',
   brown: '#A52A2A',
   burlywood: '#DEB887',
+  burgandy: '#800020',
+  burgundy: '#800020',
```
Comment (okbel) [inline lib/colors.ts:60]: Duplicated? Or 👍 ?

Comment (cond0r) [reply lib/colors.ts:60]: That's the way i found the color values are translated too (by the plugins if you choose an automatic one), so i put the translated one to test and forgot to take `burgandy` out. Used that map to convert string to hex, and check if the color is valid. Then if not to show the text, like:  https://commerce-shopify-334ih0w49-vercel-solutions-vtest314.vercel.app/product/macbook-pro later on more colors can be aded, and also based on locale to have translated versions. Should i make a commit to take that out ?

**Pull request closed without merging**